core/metacling/src/TCling.cxx

```diff
@@ -5848,8 +5848,13 @@ void* TCling::LazyFunctionCreatorAutoload(const std::string& mangled_name) {
    //  function name.
    //
 
-   if (!strncmp(name.c_str(), "public: __thiscall ", sizeof("public: __thiscall ")-1)) {
-      name.erase(0, sizeof("public: __thiscall ")-1);
+   std::string::size_type pos = name.find("__thiscall ");
```

Comment:
Can we transform this into
`for (auto thingToRemove: {"__thiscall ", "__cdecl ", "typeinfo for ", "vtable for "}) {`
?